aggregator/src/diagnostic.rs

```diff
@@ -0,0 +1,123 @@
+//! Writing diagnostic files to disk.
+
+use anyhow::Context;
+use derivative::Derivative;
+use janus_messages::{AggregationJobId, ReportMetadata};
+use std::{fmt::Debug, fs::File, io::Write, path::Path, time::SystemTime};
+use uuid::Uuid;
+
+/// Represents an illegal attempt to mutate an aggregation job.
+#[derive(Derivative, Clone)]
+#[derivative(Debug)]
+pub struct AggregationJobInitForbiddenMutationEvent {
+    /// The ID of the aggregation job.
+    pub aggregation_job_id: AggregationJobId,
```

Comment:
It would be useful to stash the TaskId as well.

Reply:
Done!